### ISSUE TITLE
4.7 support add supportsBackOffice function

### ISSUE THREAD

**Issue by eileenmcnaughton** [iATSPayments/com.iatspayments.civicrm#100]
In 4.7 the supportBackOffice function determines which processors are offered backoffice. The parent class enables credit card but not dd - so you need to add a supportsBackOffice function that returns true.

NOTE in 4.7 the form switching works  - so if it loads credit card fields and you select direct debit it will load the dd fields - although there is some other method over-ridery to control it.

@davecivicrm this is what you need to add to get it to load - but just return TRUE

```
 /**
   * Are back office payments supported.
   *
   * e.g paypal standard won't permit you to enter a credit card associated
   * with someone else's login.
   * The intention is to support off-site (other than paypal) & direct debit but that is not all working yet so to
   * reach a 'stable' point we disable.
   *
   * @return bool
   */
  protected function supportsBackOffice() {
    if ($this->_paymentProcessor['billing_mode'] == 4 || $this->_paymentProcessor['payment_type'] != 1) {
      return FALSE;
    }
    else {
      return TRUE;
    }
  }
```

**Comment by adixon** [comment]
Okay, just trying this out on 4.7. I've added the supportsBackOffice method, returning TRUE.

Is that supposed to enable processors using that class to magically appear on the backend "Submit Credit Card Contribution" form? Or is there something else required here? Because it's not ...

**Comment by eileenmcnaughton** [comment]
yeah  - from memory I thought that returning true here would be enough in 4.7....
